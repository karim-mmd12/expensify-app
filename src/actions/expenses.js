import { v4 as uuid } from 'uuid';
// ADD_EXPENSE
// Takes an object and generate an action from it
export const addExpense = (
    {
        description = '',
        note = '',
        amount = '',
        createdAt = 0
    } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

// REMOVE_EXPENSE
export const removeExpense = (
    {
        id
    } = {}
) => ({
    type: 'REMOVE_EXPENSE',
    id
});

// Edit_EXPENSE
export const editExpense = (
    id,
    updates
) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});