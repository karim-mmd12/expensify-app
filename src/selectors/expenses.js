import moment from 'moment';

/* Get visible expenses filtered by date and text */
export default (expenses, { text, sortBy, startDate, endDate }) => {
    console.log('expenses', expenses);
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? createdAtMoment.isSameOrAfter(startDate, 'day') : true;
        const endDateMatch = endDate ? createdAtMoment.isSameOrBefore(endDate, 'day') : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        /* console.log(`
        ${startDateMatch}
        ${startDateMatch}
        ${startDateMatch}`); */
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return moment(a.createdAt).isBefore(moment(b.createdAt), 'day') ? 1 : -1;
        } else if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    });
}