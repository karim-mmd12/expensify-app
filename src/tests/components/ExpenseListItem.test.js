import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('should render the ExpenseListItem component correctly', () => {
    expect(shallow(<ExpenseListItem {...expenses[0]} />)).toMatchSnapshot();
});