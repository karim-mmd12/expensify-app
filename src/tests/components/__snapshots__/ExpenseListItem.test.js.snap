// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ExpenseListItem component correctly 1`] = `
<div>
  <Link
    to="/edit/84c57dae-13af-465a-bbfe-3f4ab8e9a579"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    4500
     - 
    January 1, 1970
  </p>
</div>
`;
